airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/oauth.py

```diff
@@ -94,8 +96,11 @@ def get_refresh_request_body(self) -> Mapping[str, Any]:
     def get_token_expiry_date(self) -> pendulum.DateTime:
         return self._token_expiry_date
 
-    def set_token_expiry_date(self, value: pendulum.DateTime):
-        self._token_expiry_date = value
+    def set_token_expiry_date(self, value: Union[str, int]):
```

Comment:
can you add unit tests to cover the interface change? We should test all valid input types and branches